diktat-rules/src/main/kotlin/org/cqfn/diktat/ruleset/rules/chapter1/IdentifierNaming.kt

```diff
@@ -371,7 +371,9 @@ class IdentifierNaming(configRules: List<RulesConfig>) : DiktatRule(
         // We don't need to ask subclasses to rename superclass methods
         if (!node.isOverridden()) {
             // if function has Boolean return type in 99% of cases it is much better to name it with isXXX or hasXXX prefix
+            @Suppress("COLLAPSE_IF_STATEMENTS")
             if (functionReturnType != null && functionReturnType == PrimitiveType.BOOLEAN.typeName.asString()) {
+                @Suppress("COLLAPSE_IF_STATEMENTS")
```

Comment:
```suggestion
            @Suppress("COLLAPSE_IF_STATEMENTS")
            if (functionReturnType != null && functionReturnType == PrimitiveType.BOOLEAN.typeName.asString()) {
```
You can add suppress only on topmost node, because our logic looks at all parent nodes

Reply:
I tried, but it didn't worked. Open the issue https://github.com/cqfn/diKTat/issues/824